packages/graphiql/vite.config.ts

```diff
@@ -0,0 +1,48 @@
+import { defineConfig } from 'vite';
+import reactRefresh from '@vitejs/plugin-react-refresh';
+import visualizer from 'rollup-plugin-visualizer';
+import path from 'path';
+import express from 'express';
+const { graphqlHTTP } = require('express-graphql');
+const schema = require('./test/schema');
+const { schema: badSchema } = require('./test/bad-schema');
+const Router = require('router');
+
+export default defineConfig({
+  plugins: [
+    // reactRefresh(),
```

Comment:
Here's an explanation of how library mode works:

https://vitejs.dev/guide/build.html#library-mode

we need to use this, and specify `react` and `react-dom` as externals, so the shipped UMD bundle has parity and doesn't break existing implementations.

we need the demo script for development to use this UMD global, and not use `imports`, because the former is more widely used

Reply:
I think it will help you a lot to just take some time to study the webpack configuration, and understand how users are consuming this bundle, and try to re-create the same with vite